src/domain/shared/utils/scroll-utils/ScrollToElement.ts

```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+
+interface UseScrollToElementProps {
+  enabled: boolean;
+  elementId: string | undefined;
+}
+
+/**
+ * Executes window.scrollTo as soon as React adds an element to the DOM with the ID specified.
+ * Usage: useScrollToElement({ elementId, enabled }) will return a function that should be executed every time an element
+ * is added to the DOM for example in a map of components retrieved from the database.
+ * if elementId matches the elementIdToScrollTo specified in the hook a window.scrollTo will be triggered.
+ * @returns A function to be used as the ref={el => addElement(elId, el) } of the elements that we may want to scroll to.
+ */
+const useScrollToElement = ({ enabled, elementId: elementIdToScrollTo }: UseScrollToElementProps) => {
+  if (!elementIdToScrollTo) {
+    enabled = false;
+  }
+
+  const [alreadyScrolled, setAlreadyScrolled] = useState(!enabled);
```

Comment:
The logic seems off - if it's enabled then you are already scrolled and nothing will happen?